addons/udes_stock/models/product_product.py

```diff
@@ -134,3 +135,61 @@ def write(self, vals):
         check_upper_case_validation(self._name, self.env.user, vals)
         return super().write(vals)
 
+    def get_barcode(self):
+        """
+        Get barcode of the product depending on configuration
+        """
+        self.ensure_one()
+        if self.u_multiple_barcodes:
+            barcode = self.u_barcode_ids and self.u_barcode_ids[0].name or None
+        else:
+            barcode = self.barcode
+        return barcode
+
+    def get_list_barcodes(self):
+        """
+        Get list of barcodes for a single product.
+        """
+        self.ensure_one()
+        if self.u_multiple_barcodes:
+            list_barcodes = self.u_barcode_ids.mapped("name")
+        else:
+            # Always returning a list, even when only one barcode is in it.
+            list_barcodes = [self.barcode]
+        return list_barcodes
+
+    @api.model
+    def get_product_from_barcode(self, barcode):
```

Comment:
How about `get_by_barcode`?  It's a method on the `Product` model, so it should be obvious that it will return a product?